### DISCUSSION TITLE
Use artillery to run load test multiple API's

### DISCUSSION THREAD

**Discussion by snehalata20** [artilleryio/artillery#1134] [Artillery]
Hello,

I have started using artillery tool for load testing purpose.
Since I am new to this tool i have few below questions.

1. If I want to configure load testing for multiple  api's how i need to configure this. 
  Should i create different yaml file for each api in same project or there is any other way to achive this?
 2. I have created load testing for one of the api. And I found that when i try to increase load artillery is giving me timeout issues.
  I am trying to understand why i am getting this timeout issue. and trying to compare the calls in azure insights where we log our api call details.
As per my observation number of api's in artillery report is not matchin with azure insights. Also min , max values are also not matching.
Also in azure insights i got few errors with 408 error code which i didnt get in artillery report.

Can someone please help me to understand this issue. I have stuck here and not able to analyse the issues.

**Comment by hassy** [comment]
hi @snehalata20 

1) Multiple files for testing multiple APIs is the way to go. If you want to test multiple deployments of the same API, eg. a dev and a staging environment, then the environments feature may be helpful: https://artillery.io/docs/guides/guides/test-script-reference.html#Environments

2) Latency numbers won't match precisely as Azure Insights will only show server processing time, whereas Artillery tracks latency which also includes network time and time spent in any intermediate layers, e.g. proxies & loadbalancers.
